sdk/state/open_test.go

```diff
@@ -336,76 +315,36 @@ func TestChannel_OpenTx(t *testing.T) {
 	// channel.openTx() when there is an practical way of doing that added to
 	// txnbuild.
-	assert.Equal(t, []xdr.DecoratedSignature{{Hint: [4]byte{2, 2, 2, 2}, Signature: []byte{2}}}, formationTx.Signatures())
-}
-
-func TestConfirmOpen_checkForExtraSignatures(t *testing.T) {
-	localSigner := keypair.MustRandom()
-	remoteSigner := keypair.MustRandom()
-	localEscrowAccount := &EscrowAccount{
-		Address:        keypair.MustRandom().FromAddress(),
-		SequenceNumber: int64(101),
-	}
-	remoteEscrowAccount := &EscrowAccount{
-		Address:        keypair.MustRandom().FromAddress(),
-		SequenceNumber: int64(202),
-	}
-	senderChannel := NewChannel(Config{
-		NetworkPassphrase:   network.TestNetworkPassphrase,
-		Initiator:           true,
-		LocalSigner:         localSigner,
-		RemoteSigner:        remoteSigner.FromAddress(),
-		LocalEscrowAccount:  localEscrowAccount,
-		RemoteEscrowAccount: remoteEscrowAccount,
-	})
-	receiverChannel := NewChannel(Config{
-		NetworkPassphrase:   network.TestNetworkPassphrase,
-		Initiator:           false,
-		LocalSigner:         remoteSigner,
-		RemoteSigner:        localSigner.FromAddress(),
-		LocalEscrowAccount:  remoteEscrowAccount,
-		RemoteEscrowAccount: localEscrowAccount,
-	})
-
-	p := OpenParams{
-		ObservationPeriodTime:      time.Minute,
-		ObservationPeriodLedgerGap: 2,
-		Asset:                      "native",
-		ExpiresAt:                  time.Date(2020, 1, 2, 3, 4, 5, 6, time.UTC),
-	}
-
-	m, err := senderChannel.ProposeOpen(p)
-	require.NoError(t, err)
-
-	m.CloseSignatures = append(m.CloseSignatures, xdr.DecoratedSignature{Signature: randomByteArray(t, 10)})
-
-	// Extra signature should cause error when receiver confirms
-	_, err = receiverChannel.ConfirmOpen(m)
-	assert.Equal(t, OpenAgreement{}, receiverChannel.openAgreement)
-	require.EqualError(t, err, "input open agreement has too many signatures, has declaration: 2, close: 3, formation: 2, max of 2 allowed for each")
-
-	// Remove extra signature, now should succeed
-	m.CloseSignatures = m.CloseSignatures[0:1]
-	m, err = receiverChannel.ConfirmOpen(m)
-	require.NoError(t, err)
-
-	// Adding extra signature should fail when sender confirms
-	m.DeclarationSignatures = append(m.DeclarationSignatures, xdr.DecoratedSignature{Signature: randomByteArray(t, 10)})
-	_, err = senderChannel.ConfirmOpen(m)
-	require.EqualError(t, err, "input open agreement has too many signatures, has declaration: 3, close: 2, formation: 2, max of 2 allowed for each")
-
-	// Remove extra signature, now should succeed
-	m.DeclarationSignatures = m.DeclarationSignatures[0:2]
-	m, err = senderChannel.ConfirmOpen(m)
-	require.NoError(t, err)
+	assert.ElementsMatch(t, []xdr.DecoratedSignature{
+		{Hint: localSigner.Hint(), Signature: []byte{2}},
+		{Hint: remoteSigner.Hint(), Signature: []byte{5}},
+		xdr.NewDecoratedSignatureForPayload([]byte{3}, remoteSigner.Hint(), declTxHash[:]),
+		xdr.NewDecoratedSignatureForPayload([]byte{4}, remoteSigner.Hint(), closeTxHash[:]),
```

Comment:
Further up on line 318 the proposer and confirmer for the open agreement are defined as the local and remote respectively. The remote is not always the confirmer, as it depends who proposed the agreement. In this test we expect the remote to have confirmed the transaction, and therefore be providing additional signatures, simply because they're defined as the confirmer in the agreement.
https://github.com/stellar/experimental-payment-channels/blob/12893a19c8e68ca52337a3b06db040f26579f5bc/sdk/state/open_test.go#L318-L319

For reading material on how the additional signatures work I recommend reading https://github.com/stellar/experimental-payment-channels/pull/176 and [CAP-40](https://stellar.org/protocol/cap-40). The TL;DR is that there are two to three transactions that need signing for each agreement. It is unsafe for one participant to share a signature for the first transaction before everyone has signed the transactions after it. So we require that the signatures for the second transaction, and third transaction if it exists, are embedded in the first. Only the confirmer needs to do this because the confirmer can cause the proposer harm otherwise. We assume the proposer doesn't want to cause themselves harm so they don't need to do this.